### ISSUE TITLE
Support for caching_sha2_password authentication plugin

### ISSUE THREAD

**Issue by zhfahan77** [Synor/synor#34]
In MySQL 8 and onwards version, `mysql_native_password` plugin will be deprecated. In order to support `caching_sha2_password` plugin, we somehow need to pass a configuration like this `default-authentication-plugin`.

[Default Authentication Plugin](https://dev.mysql.com/blog-archive/upgrading-to-mysql-8-0-default-authentication-plugin-considerations/)

Can we modify synor to take authentication configuration?

**Comment by zhfahan77** [comment]
<img width="1557" alt="CleanShot 2022-11-02 at 11 02 32@2x" src="https://user-images.githubusercontent.com/18385348/199402355-f91c31d8-e4de-4259-8002-52ceb793ac76.png">

This is the error we get while trying to run `mg:info` command

**Comment by zhfahan77** [comment]
Configuration used

```
{
  "hosts": [{ "name": "127.0.0.1", "port": 3306 }],
  "user": "user",
  "password": "password",
  "path": ["name"],
  "params": {
    "synor_migration_record_table": "synor_migration_record",
  }
}

**Comment by MunifTanjim** [comment]
Fixed in https://github.com/Synor/synor/pull/35.

Published as [`v0.7.0`](https://www.npmjs.com/package/@synor/database-mysql/v/0.7.0).

The driver was changed from `mysql` to `mysql2`.

So you'll need to also install `mysql2`.

**Comment by zhfahan77** [comment]
Thanks 🔥 it works
